app/helpers/course/assessment/answer/programming_helper.rb

```diff
@@ -123,4 +155,45 @@ def build_annotation_discussion(line_discussion_cell, discussion_topic)
 
     line_discussion_cell.inner_html = html
   end
+
+  # Return a hash of the first failing test case and its test result
+  #
+  # @param [Hash] test_cases_and_results_of_type A hash of test cases and results keyed by type
+  # @return [Hash] the failed test case and result, nil if all tests passed
+  def get_first_failed_test(test_cases_and_results_of_type)
+    test_cases_and_results_of_type.each do |test_case, test_result|
+      return {}.tap { |h| h[test_case] = test_result } if test_result && !test_result.passed?
```

Comment:
There is no need to tap here - `return { test_case: test_result } if test_result && !test_result.passed?`